core/trino-main/src/main/java/io/trino/sql/planner/ConnectorExpressionTranslator.java

```diff
@@ -13,86 +13,163 @@
+                return Optional.of(literalEncoder.toExpression(session, ((Constant) expression).getValue(), expression.getType()));
             }
 
             if (expression instanceof FieldDereference) {
                 FieldDereference dereference = (FieldDereference) expression;
-                return new SubscriptExpression(translate(session, dereference.getTarget()), new LongLiteral(Long.toString(dereference.getField() + 1)));
+                return translate(session, dereference.getTarget())
+                        .map(base -> new SubscriptExpression(base, new LongLiteral(Long.toString(dereference.getField() + 1))));
+            }
+
+            if (expression instanceof Call) {
+                return translateCall((Call) expression);
             }
 
-            throw new UnsupportedOperationException("Expression type not supported: " + expression.getClass().getName());
+            return Optional.empty();
+        }
+
+        protected Optional<Expression> translateCall(Call call)
+        {
+            if (call.getFunctionName().getCatalogSchema().isPresent()) {
+                return Optional.empty();
+            }
+            QualifiedName name = QualifiedName.of(call.getFunctionName().getName());
+            List<TypeSignature> argumentTypes = call.getArguments().stream()
+                    .map(argument -> argument.getType().getTypeSignature())
+                    .collect(toImmutableList());
+            ResolvedFunction resolved = plannerContext.getMetadata().resolveFunction(session, name, TypeSignatureProvider.fromTypeSignatures(argumentTypes));
+
+            // TODO Support ESCAPE character
+            if (LIKE_PATTERN_FUNCTION_NAME.equals(resolved.getSignature().getName()) && call.getArguments().size() == 2) {
+                return translateLike(call.getArguments().get(0), call.getArguments().get(1));
+            }
+
+            FunctionCallBuilder builder = FunctionCallBuilder.resolve(session, plannerContext.getMetadata())
+                    .setName(name);
+            for (int i = 0; i < call.getArguments().size(); i++) {
+                Type type = resolved.getSignature().getArgumentTypes().get(i);
+                Expression expression = ConnectorExpressionTranslator.translate(session, call.getArguments().get(i), plannerContext, variableMappings, literalEncoder);
+                builder.addArgument(type, expression);
+            }
+            return Optional.of(builder.build());
+        }
+
+        protected Optional<Expression> translateLike(ConnectorExpression value, ConnectorExpression pattern)
+        {
+            Optional<Expression> translatedValue = translate(session, value);
+            Optional<Expression> translatedPattern = translate(session, pattern);
+            if (translatedValue.isPresent() && translatedPattern.isPresent()) {
+                return Optional.of(new LikePredicate(translatedValue.get(), translatedPattern.get(), Optional.empty()));
```

Comment:
@martint  Shouldn't this use `LikeFunctions#LIKE_PATTERN_FUNCTION_NAME`, so that we don't have to re-run `DesugarLike`?

cc @wendigo